server/src/services/database.service.ts

```diff
@@ -1,163 +1,134 @@
 import { Inject, Injectable } from '@nestjs/common';
+import semver from 'semver';
+import { POSTGRES_VERSION_RANGE, VECTORS_VERSION_RANGE, VECTOR_VERSION_RANGE } from 'src/constants';
+import { getVectorExtension } from 'src/database.config';
 import {
   DatabaseExtension,
   DatabaseLock,
+  EXTENSION_NAMES,
   IDatabaseRepository,
-  VectorExtension,
   VectorIndex,
-  extName,
 } from 'src/interfaces/database.interface';
 import { ILoggerRepository } from 'src/interfaces/logger.interface';
-import { Version, VersionType } from 'src/utils/version';
 
 @Injectable()
 export class DatabaseService {
-  private vectorExt: VectorExtension;
-  minPostgresVersion = 14;
-  minVectorsVersion = new Version(0, 2, 0);
-  vectorsVersionPin = VersionType.MINOR;
-  minVectorVersion = new Version(0, 5, 0);
-  vectorVersionPin = VersionType.MAJOR;
-
   constructor(
     @Inject(IDatabaseRepository) private databaseRepository: IDatabaseRepository,
     @Inject(ILoggerRepository) private logger: ILoggerRepository,
   ) {
     this.logger.setContext(DatabaseService.name);
-    this.vectorExt = this.databaseRepository.getPreferredVectorExtension();
   }
 
   async init() {
-    await this.assertPostgresql();
+    const version = await this.databaseRepository.getPostgresVersion();
+    const current = semver.coerce(version);
+    if (!current || !semver.satisfies(current, POSTGRES_VERSION_RANGE)) {
+      throw new Error(
+        `Invalid PostgreSQL version. Found ${version}, but needed ${POSTGRES_VERSION_RANGE}. Please use a supported version.`,
+      );
+    }
+
     await this.databaseRepository.withLock(DatabaseLock.Migrations, async () => {
-      await this.createVectorExtension();
-      await this.updateVectorExtension();
-      await this.assertVectorExtension();
```

Comment:
Why did you inline all of this? It looks really messy this way.

Reply:
I removed all the logging and consolidated the shared variables, so it looks a lot better now. Can you take another look?

Reply:
Much better now, thanks.